vm-systemd/qubes-qrexec-agent.service

```diff
@@ -1,6 +1,6 @@
 [Unit]
 Description=Qubes remote exec agent
-After=qubes-dvm.service
+After=qubes-dvm.service xendriverdomain.service
```

Comment:
This is what `qubes.WaitForSession` service is for. And it basically wait for X session to be fully started (but can be extended to other things if needed).